Documentation/cli-installation-scenarios.md

```diff
@@ -0,0 +1,266 @@
+
+The native installers are:
+
+| Platform            	| Installer        	| Description                                                                                                                                                                                                                                        	|
+|---------------------	|------------------	|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------	|
+| Windows             	| Bundle installer 	| Windows native installation technology is what we use to install stuff.                                                                                                                                                                            	|
+| Ubuntu 14.04/Debian 	| apt-get feed     	| We provide the Debian package on a hosted feed. This means that the users need to add the feed, add keys and then use apt-get to install the CLI tools.                                                                                            	|
+| OS X                	| PKG              	| PKGs are files that installer(1) on OS X uses to install software. One big drawback is that we cannot bundle depencencies for CoreCLR and CoreFX within this installation method, unless those dependencies are available in binary form already.  	|
+| OS X                	| Homebrew         	| Though not really "native" as in "available OOB on the system", homebrew has risen as a *de facto* package manager for OS X.                                                                                                                       	|
+| CentOS/RH           	| RPM              	| Though it currently doesn't exist, we should look into enabling RPMs for CentOS/RH-based distributions. This would bring it to par with Debian ones.                                                                                               	|
+
+
+## Installation script
+This approach is a shell one-liner that downloads an installation script and runs it. The installation script will download the latest zip/tarball (depending on the script used) and will unpack it to a given location. After that, the script will print out what needs to be set for the entire CLI to work (env variables, $PATH modification etc.).
+
+This install covers the following main scenario: 
+
+* Local installation on a dev machine
+* Acquiring tools on a CI build
+
+  
+The features the script needs to support/have are:
+* Support for dev and stable channel
```

Comment:
Can we make Channel a string that is passed in which will default to Preview until we have a Production release, then will default to Production